pkg/reconciler/managed/reconciler.go

```diff
@@ -187,6 +188,46 @@ type ExternalConnecter interface {
 	Connect(ctx context.Context, mg resource.Managed) (ExternalClient, error)
 }
 
+// An ExternalDisconnecter disconnect from provider.
+type ExternalDisconnecter interface {
+	// Disconnect from the provider and close an ExternalClient.
+	Disconnect(ctx context.Context) error
+}
+
+// NopDisconnect do nothing. Never returns error.
+func NopDisconnect(_ context.Context) error {
+	return nil
+}
+
+// closableExternalConnector is wrap ExternalConnecter in closableExternalConnector
+// with nop Disconnect.
+type closableExternalConnector struct {
+	c ExternalConnecter
+	d ExternalDisconnecter
+}
+
+// Connect call ExternalConnecter's Connect
+func (c *closableExternalConnector) Connect(ctx context.Context, mg resource.Managed) (ExternalClient, error) {
+	return c.c.Connect(ctx, mg)
+}
+
+// Disconnect call ExternalDisconnecter's Disconnect
+func (c *closableExternalConnector) Disconnect(ctx context.Context) error {
+	return c.d.Disconnect(ctx)
+}
+
+// NewExternalConnectorClosable wraps ExternalConnecter in ExternalConnecterClosable
+func NewExternalConnectorClosable(c ExternalConnecter, d ExternalDisconnecter) ExternalConnecterClosable {
+	return &closableExternalConnector{c, d}
+}
+
+// An ExternalConnecterClosable produces a new ExternalClient given the supplied
+// Managed resource.
+type ExternalConnecterClosable interface {
```

Comment:
```suggestion
type ExternalConnectDisconnecter interface {
```

What do you think about this name? I think it reads a little more like the typical Go interfaces like `io.ReadWriter` or `io.ReadWriteCloser`. I slightly prefer `Disconnecter` to `Closable` because the new method is `Disconnect` rather than `Close`.

Reply:
Also, technically this should probably be `Disconnector` (i.e. ending in "tor", not "ter") but we already have a spelling mistake in `ExternalConnecter`. 😦